### PR TITLE
create server and router #9

### PR DESCRIPTION
**Pull request by Samaamro20**
- create server.js
- create router.js

### CODE REVIEW

File: src/server/server.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const http = require('http');
+const router = require('./router.js');
+require('dotenv').config()
```
Comment (jema28) [inline src/server/server.js]: You're already using env2 in another file: dotenv is the same thing so isn't not necessary to have both.

File: src/server/router.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  case 'GET /addbook':
+    handler.addbookHandler(request,response);
+    break;
+  case 'GET /profile':
+    handler.profileHandler(request,response);
+    break;
+  default:
+      handler.notFoundHandler(request,response);
+}
+ 	// if (pathname === '/' ) {
```
Comment (jema28) [inline src/server/router.js]: Please delete these comments if not being used.

File: src/server/router.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	// else if (pathname.includes('/search/')) {
+	// 	handler.searchHandler(request,response);
+	// }
+	// else
+	// {
+	// 	handler.notFoundHandler(request,response);
+	// }
+}
+module.exports=router;
+// const payload = {
```
Comment (jema28) [inline src/server/router.js]: Please delete these comments also, they shouldn't be in this pull request.

**Review by jema28 — CHANGES_REQUESTED** [review]

**Review by amanshawar — APPROVED** [review]

**Review by Hadeel-Salamin — APPROVED** [review]

File: src/server/handler.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+const requester = require('request');
+const fs = require('fs');
+const cookie = require('cookie');
+const jwt = require('jsonwebtoken');
+
+//const getData=require('./queries/get.js');
+//const setData=require('./queries/set.js');
+//const check=require('./queries/check.js');
+//const alert = require('alert-node');
+const SECRET = '123book123';
```
Comment (jema28) [inline src/server/handler.js]: Secret shouldn't be here!

File: src/server/handler.js [hunk 5]
```diff
@@ -0,0 +1,217 @@
+const path = require('path');
+const querystring = require('querystring');
+const requester = require('request');
+const fs = require('fs');
+const cookie = require('cookie');
+const jwt = require('jsonwebtoken');
+
+//const getData=require('./queries/get.js');
```
Comment (jema28) [inline src/server/handler.js]: Look at the comments and see if you can get rid of them.

File: src/server/handler.js [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+}
+
+
+//-----------------------------------------------------------------------------
+
+const logoutHandler = (request, response) => {
+	response.writeHead(302, {
+		'Location': '/',
+		'Set-Cookie': `logged_in=${0}; Max-Age=0`
```
Comment (jema28) [inline src/server/handler.js:63]: Why do you need to put 0 in $`` here?

File: src/server/handler.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
+const path = require('path');
+const querystring = require('querystring');
```
Comment (jema28) [inline src/server/handler.js]: Request and cookie modules aren't being used.

File: src/server/handler.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+const fs = require('fs');
+const cookie = require('cookie');
+const jwt = require('jsonwebtoken');
+
+const SECRET = '123book123';
+const userDetails = {
+	userId: 5,
+};
+
+var token = jwt.sign(userDetails, SECRET);
```
Comment (jema28) [inline src/server/handler.js]: please change var to const or let here to keep things consistent.

**Review by jema28 — CHANGES_REQUESTED** [review]

**Review by jema28 — APPROVED** [review]

**Pull request merged**